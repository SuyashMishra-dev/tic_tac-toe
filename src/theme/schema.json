{
  "data": {
    "light": {
      "id": "T_001",
      "name": "Light",
      "colors": {
        "body": "#FFFFFF",
        "text": "#000000",
        "border_color": "#4fb472",
        "dark_blue": "#001E82",
        "light_blue": "#D8EFFE",
        "dark_green": "#0fb839",
        "my_red": "#f70808"
      },
      "fonts": {
        "font_size": {
          "heavy_font": "32px",
          "medium_font": "16px",
          "below_small_font": "14px",
          "small_font": "12px",
          "smallest_font": "10px"
        },
        "font_weight": {
          "bolder_font": 800,
          "bold_font": 700,
          "normal_font": 400
        }
      }
    },
    "dark": {
      "id": "T_002",
      "name": "Dark",
      "colors": {
        "body": "#000000",
        "text": "#FFFFFF",
        "border_color": "#d9f867",
        "dark_blue": "#c2a920",
        "light_blue": "#a59505",
        "dark_green": "#42e910",
        "my_red": "#f77c08"
      },
      "fonts": {
        "font_size": {
          "heavy_font": "32px",
          "medium_font": "16px",
          "below_small_font": "14px",
          "small_font": "12px",
          "smallest_font": "10px"
        },
        "font_weight": {
          "bolder_font": 800,
          "bold_font": 700,
          "normal_font": 400
        }
      }
    }
  }
}
