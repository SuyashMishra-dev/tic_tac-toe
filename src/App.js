import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { useTheme } from './theme/useTheme';
import { GlobalStyles } from './theme/GlobalStyles';
import GameContainer from './components/gameContainer/GameContainer';
import ChangeTheme from './components/changeTheme/ChangeTheme';

const App = () => {
  const [themeValue, setThemeValue] = useState(false);
  const { theme, themeLoaded, setMode } = useTheme('light');
  const [selectedTheme, setSelectedTheme] = useState(theme);
  useEffect(() => {
    setSelectedTheme(theme);
  }, [theme, themeLoaded]);
  useEffect(() => {
    setMode(themeValue ? 'dark' : 'light');
  }, [themeValue, setMode]);
  const changeTheme = (event) => {
    setThemeValue(!themeValue);
  }
  return (
    <>
      {themeLoaded && (
        <ThemeProvider theme={selectedTheme}>
          <GlobalStyles />
          <ChangeTheme changeTheme={changeTheme} />
          <GameContainer />
        </ThemeProvider>
      )}
    </>
  );
};

export default App;
