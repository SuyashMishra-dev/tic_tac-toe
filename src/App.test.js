import React from 'react';
import { act } from 'react-dom/test-utils';
import {shallow, configure } from 'enzyme'
import TestRenderer from 'react-test-renderer';
import { renderHook } from '@testing-library/react-hooks'
import { render, fireEvent, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import * as themes from './theme/schema.json';
import { checkWinner } from './utils/checkResult';
import Board from './components/gameBoard/GameBoard';
import Game from './components/gameContainer/GameContainer';
import App from './App';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

const renderComponent = ({ theme }) => act(() => {
  render(
      <ThemeProvider theme={theme}>
        <App />
      </ThemeProvider>,
  );
});


describe('test for App screen', () => {
  it('renders correctly when there are no items', async () => {
    // when
    const tree = await TestRenderer.create(<App />).toJSON();
    // then
    expect(tree).toMatchSnapshot();
  });

  it('test checkbox clicks', async () => {
    // when
    const { waitForNextUpdate } = renderHook(
      () => act(() => { renderComponent({ theme: themes.data.light }); }),
    );
    // then
    const checkbox = screen.getByTestId('change');
    waitForNextUpdate();
    expect(checkbox.checked).toEqual(false)
    fireEvent.click(checkbox)
    waitForNextUpdate();
    expect(checkbox.checked).toEqual(true)
  });

  it('check checkWinner function is working properly', () => {
    // when
    const originalArray = ['O', 'X', 'X', 'O', 'O', 'X', null, null, null];
    const expected = null;
    const sortResult = checkWinner(originalArray);
    // then
    expect(sortResult).toEqual(expected);
  });

  it('test', async () => {
    // when
    const { waitForNextUpdate } = renderHook(
      () => act(() => { renderComponent({ theme: themes.data.light }); }),
    );
    // then
    const Button = screen.queryByTestId('Undo');
    fireEvent.click(Button)
    waitForNextUpdate();
    expect(Button).toBeTruthy()
  });

  it('renders without crashing for squares', () => {
    let squares = Array(9).fill(null)
    shallow(<Board squares={squares}/>);
  });


  it('render without crashing for game board', () => {
    shallow(<Game />);
  });

});