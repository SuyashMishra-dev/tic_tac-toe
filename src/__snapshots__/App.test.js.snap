// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for App screen renders correctly when there are no items 1`] = `null`;
